server.js

```diff
@@ -0,0 +1,49 @@
+//Initialize 
+const express = require('express')
+const cors = require("cors")
+const app = express()
+let users = [
+    { _id: "u1", username: "user1", password: "pw1"},
+    { _id: "u2", username: "user2", password: "pw2"},
+    { _id: "u3", username: "user3", password: "pw3"},
+]
+
+//Middleware
+app.use( cors() ) 
+app.use( ( express.json() ))
+
+
+
+//Routing
+app.get('/users', (request, response) => {
+    response.json(users)
+})
+
+app.get('/users/:user', (req, res) => {
+    const {user} = req.params
+    const userBack = users.find( element => element.username === user)
```

Comment:
Fetching by ID is actually boring, so fetching by name is a nice alternative .

But in general: Fetching by ID is actually way more performant ;)